packages/geo/src/lib/geometry/geometry-form-field/geometry-form-field-input.component.ts

```diff
@@ -209,15 +224,22 @@ export class GeometryFormFieldInputComponent implements OnInit, OnDestroy, Contr
    * Create a draw control and subscribe to it's geometry
    */
   private createDrawControl() {
-    this.drawControl = new DrawControl({
-      geometryType: this.geometryType,
-      layer: this.olOverlayLayer,
-      drawStyle: (olFeature: OlFeature, resolution: number) => {
-        const style = this.drawInteractionStyle;
-        this.updateDrawStyleWithDrawGuide(style, resolution);
-        return style;
-      }
-    });
+    if (this.geometryType) {
```

Comment:
I think the createDrawControl in ngOnInit could be harmlessly removed and the check for a defined geometryType moved to the geometryType setter. This would remove the need for a check here.

Reply:
So no tool would be activated by default? Because if there is a geometryType defined on init the tool should be activated I think

Reply:
ngOnInit is invoked after the inputs have been resolved so there should still be at least one control active.

Reply:
I tried but it knida breaks the logic, there is a condition in the geometry type setter which returns when onInit is not completed

Reply:
Should I keep as it is or change that also

Reply:
I think it could be safely removed from the geometryType setter. I have been working on a version of this component that completely removes the need for the "ready" flag. I might push it later but, for now, let's remove it from the setter only, if possible.

Reply:
It does not simplify things a lot I fear... Draw control cannot be created before overlay layer is added to the map

Reply:
If I do some kind of default vector layer is hooked to the draw control and the first geometry is added twice